src/cloudflare/internal/images-api.ts

```diff
@@ -104,6 +138,72 @@ class ImageTransformerImpl implements ImageTransformer {
 
     return new TransformationResultImpl(response);
   }
+
+  private consume(): void {
+    if (this.consumed) {
+      throw new ImagesErrorImpl(
+        'IMAGES_TRANSFORM_ERROR 9525: ImageTransformer consumed; you may only call .output() or draw a transformer once',
+        9525
+      );
+    }
+
+    this.consumed = true;
+  }
+
+  private async serializeTransforms(body: FormData): Promise<void> {
+    const transforms: (TargetedTransform | DrawCommand)[] = [];
+
+    // image 0 is the canvas, so the first draw_image has index 1
+    let drawImageIndex = 1;
+    function appendDrawImage(blob: Blob): number {
+      body.append('draw_image', blob);
+      return drawImageIndex++;
+    }
+
+    async function walkTransforms(
+      targetImageIndex: number,
+      imageTransforms: (ImageTransform | DrawTransformer)[]
+    ): Promise<void> {
+      for (const transform of imageTransforms) {
+        if (!isDrawTransformer(transform)) {
+          // Simple transformation - we just have to tell the backend to run it
+          // against this image
+          transforms.push({
+            imageIndex: targetImageIndex,
+            ...transform,
+          });
+        } else if (isDrawTransformer(transform)) {
```

Comment:
Since line 168 also checks for it

```suggestion
        } else {
```